### ISSUE TITLE
Idea: dirty input value state

### ISSUE THREAD

**Issue by nealfwilliams** [paypal/downshift#562]
**Problem description**
Something that would be useful is another property in the state that tracks if the input has been edited since the last selection.  

The use case for this is a select field with autocomplete.  Upon opening the dropdown, all the options should be shown, even if the input has a value.   Filtering using the input value doesn't start happening until the user starts typing (ie the input value is "dirty").  This property could also be used to control whether typing replaces the current input value or not.

**Suggested solution**:
Basically just set the boolean to true when the input changes and set it to false after a selection is made or the input loses focus. 

If you agree this would be a useful feature, I can submit a PR.
